src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy

```diff
@@ -81,7 +81,39 @@ class GitVersionPluginTests extends Specification {
         git.runGitCommand("add", ".")
         git.runGitCommand("commit","-m", "'initial commit'")
         git.runGitCommand("tag", "-a", "1.0.0", "-m", "1.0.0")
-        println(projectDir)
+
+        when:
+        // will build the project at projectDir
+        BuildResult buildResult = with('printVersion').build()
+
+        then:
+        buildResult.output.contains(":printVersion\n1.0.0\n")
+    }
+
+    def 'git describe works when using worktree' () {
+        given:
+        File rootFolder = temporaryFolder
+        projectDir = Files.createDirectories(rootFolder.toPath().resolve('level1/hotfix')).toFile()
+        File initialDir = Files.createDirectories(rootFolder.toPath().resolve('level1/level2')).toFile()
+        buildFile = new File(initialDir, 'build.gradle')
+        buildFile.createNewFile()
+        buildFile << '''
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            version gitVersion()
+        '''.stripIndent()
+        new File(initialDir, 'settings.gradle').createNewFile()
+        File initialGitIgnoreFile = new File(initialDir, ".gitignore")
+        initialGitIgnoreFile.createNewFile()
+        initialGitIgnoreFile << '.gradle\n'
+        Git git = new Git(initialDir, true)
+        git.runGitCommand("init", initialDir.toString())
+        git.runGitCommand("add", ".")
+        git.runGitCommand("commit","-m", "'initial commit'")
+        git.runGitCommand("tag", "-a", "1.0.0", "-m", "1.0.0")
+        git.runGitCommand("branch", "hotfix")
+        git.runGitCommand("worktree", "add", "../hotfix", "hotfix")
```

Comment:
I think `git worktree add` only takes a single path argument. I'm not sure I understand what this command is supposed to be doing.

I'd also prefer if we avoided names with specific connotations like "hotfix". There's nothing specific to "hotfixes" about worktree support.

Reply:
The command creates a worktree at `../hotfix` and checks out the `hotfix` branch. 
I fixed my code to use a more neutral name.